pkg/config/security/security.go

```diff
@@ -205,6 +205,18 @@ var ValidCipherSuites = sets.New(
 	"DES-CBC3-SHA",
 )
 
+// ValidECDHCurves contains a list of all ecdh curves supported in MeshConfig.MeshExternal_TLS.ecdhCurves
+// Source:
+// https://github.com/google/boringssl/blob/3743aafdacff2f7b083615a043a37101f740fa53/ssl/ssl_key_share.cc#L302-L309
+var ValidECDHCurves = sets.New(
```

Comment:
From Istio we will not set any default value because of this, and would let envoy choose the default. However if the control plane wants to choose a specific curve against letting envoy choose the default, we allow this config option. Did I miss any supported curve here in the validation, which will be otherwise allowed by envoy?

Reply:
Well, you are looking at a particular commit of boringssl. Curves can be added and removed at any time (a vulnerability found, US government adds it to FIPS approved list, etc). It depends on how envoy was built (which version of boringssl/openssl/fips was used). I am worried this list can become actively harmful later, because users will not be able to use the right curve.

I think the list of curves should not be built into code. It needs to be vendor-specific (US companies might want FIPS list, while the rest of the world is fine with Wireguard's CHACHA) as well as Envoy-build specific (so that we can transition between Envoy versions). 

Sending a wrong curve I believe will lead Envoy deny the entire SDS resource.

Reply:
@kyessenov : Would it be better if I remove the validation from Istio, and leave the validation responsibility to envoy?

Reply:
Yes, I'm fine with no validation. Basically, half-working validation is more harmful than no validation.

Reply:
+1 for warning, but noone should by typing manually crypto-related settings :)
They should be in gitops.

Reply:
Changed the errors to warnings